formspree/app.py

```diff
@@ -2,18 +2,31 @@
 from flask import g
 from flask.ext.sqlalchemy import SQLAlchemy
 from flask.ext.login import LoginManager, current_user
-import fakeredis, redis
+import redis
 import settings
 
+
 DB = SQLAlchemy()
-if settings.TESTING:
-    REDIS = fakeredis.FakeStrictRedis()
-else:
-    REDIS = redis.StrictRedis.from_url(settings.REDIS_URL)
+_redis = None
+
+
+def REDIS():
+    ''' function to acquire the global redis connection '''
+    global _redis
+    if not _redis:
+        _redis = get_redis()
+    return _redis
```

Comment:
This will require we add the with block around each test, or use a decorator for each test. Seems like a pain just to avoid use of a singleton. Why is a global better than a singleton?